knapsack-highspy/main.py

```diff
@@ -0,0 +1,136 @@
+"""
+Template for working with highspy.
+"""
+
+import argparse
+import json
+import sys
+import time
+from importlib.metadata import version
+from typing import Any
+
+import highspy
+
+
+def main() -> None:
+    """Entry point for the template."""
+
+    parser = argparse.ArgumentParser(description="Solve problems with highspy.")
+    parser.add_argument(
+        "-input",
+        default="",
+        help="Path to input file. Default is stdin.",
+    )
+    parser.add_argument(
+        "-output",
+        default="",
+        help="Path to output file. Default is stdout.",
+    )
+    parser.add_argument(
+        "-duration",
+        default=30,
+        help="Max runtime duration (in seconds). Default is 30.",
+        type=int,
+    )
+    args = parser.parse_args()
+
+    # Read input data, solve the problem and write the solution.
+    input_data = read_input(args.input)
+
+    log("Solving knapsack problem:")
+    log(f"  - items: {len(input_data.get('items', []))}")
+    log(f"  - capacity: {input_data.get('weight_capacity', 0)}")
+    log(f"  - max duration: {args.duration} seconds")
+
+    solution = solve(input_data, args.duration)
+    write_output(args.output, solution)
+
+
+def solve(input_data: dict[str, Any], duration: int) -> dict[str, Any]:
+    """Solves the given problem and returns the solution."""
+
+    start = time.time()
+
+    # Creates the solver.
+    h = highspy.Highs()
```

Comment:
Would call it `solver` instead of `h`, as the variable is used many times and a single letter is not as clear.

Reply:
Yeah, agreed. This was a leftover of the example I started from. I changed it.